### ISSUE TITLE
[BUG] Firefox fails to click with Kendo UI Bootstrap, element is not enabled

### ISSUE THREAD

**Issue by rolandz83** [microsoft/playwright#11600]
**Context:**
## System:
 - OS: Windows 10 10.0.17763
 - Memory: 8.81 GB / 15.88 GB
## Binaries:
 - Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
 - npm: 8.3.2 - C:\Program Files\nodejs\npm.CMD
## npmPackages:
 - playwright: ^1.18.0 => 1.18.0

**Code Snippet**

```ts
import { test, expect, Page} from '@playwright/test';

test("Demo Kendo UI with firefox", async({browser}) => {
    let page : Page = await browser.newPage();

    await page.goto('https://demos.telerik.com/kendo-ui/dropdownlist/index');
    await page.waitForLoadState('networkidle');

    await page.locator('span:has(input#categories)').click();
    await page.locator('ul[id$="_listbox"]:visible >> li:has-text("Grains/Cereals")').click();

    await page.locator('span:has(input#products)').click();
    await page.locator('ul[id$="_listbox"]:visible >> li:has-text("Ravioli Angelo")').click();

    await page.locator('span:has(input#shipTo)').click();
    await page.locator('.k-list-content:visible >> li:has-text("Torikatu 38")').first().click();
    
    await expect(page.locator('span:has(input#categories)')).toHaveText('Grains/Cereals');
    await expect(page.locator('span:has(input#products)')).toHaveText('Ravioli Angelo');
    await expect(page.locator('span:has(input#shipTo)')).toHaveText('Torikatu 38');
});
```

**Describe the bug**

Hello PW team! 
Running the above code works great with chromium and webkit 
however firefox is having difficulty clicking. sometimes it will work, offer it will fail to click on any one these actions,
for instance in this case I got the following when inspecting... it seems like another list item gets in the way

```
waiting for selector "ul[id$="_listbox"]:visible >> li:has-text("Grains/Cereals")"
selector resolved to visible <li tabindex="-1" role="option" unselectable="on" cl…>…</li>
attempting click action
waiting for element to be visible, enabled and stable
element is not stable - waiting...
element is visible, enabled and stable
scrolling into view if needed
done scrolling
performing click action
<div class="demo-section k-content">…</div> from <main role="main" class="kd-jquery">…</main> subtree intercepts pointer events
retrying click action, attempt #1
waiting for element to be visible, enabled and stable
element is not visible - waiting...
```

I tried adding delays and event compensation for the network calls that happens when the first drop down is clicked, but nothing seems to help. Thank you for taking time to look at this, firefox has been the head-scratcher for a long time :)

**Comment by aslushnikov** [comment]
Hey @rolandz83, I just ran this test 100 times and it passed all 100 times for me.

I did it with a pristine folder:

```bash
$ mkdir tmp && cd tmp
$ npm init -y && npm i @playwright/test && npx playwright install
```

Here I put `a.spec.ts` which is your code exactly:

```ts
import { test, expect, Page} from '@playwright/test';

test("Demo Kendo UI with firefox", async({browser}) => {
    let page : Page = await browser.newPage();

    await page.goto('https://demos.telerik.com/kendo-ui/dropdownlist/index');
    await page.waitForLoadState('networkidle');

    await page.locator('span:has(input#categories)').click();
    await page.locator('ul[id$="_listbox"]:visible >> li:has-text("Grains/Cereals")').click();

    await page.locator('span:has(input#products)').click();
    await page.locator('ul[id$="_listbox"]:visible >> li:has-text("Ravioli Angelo")').click();

    await page.locator('span:has(input#shipTo)').click();
    await page.locator('.k-list-content:visible >> li:has-text("Torikatu 38")').first().click();
    
    await expect(page.locator('span:has(input#categories)')).toHaveText('Grains/Cereals');
    await expect(page.locator('span:has(input#products)')).toHaveText('Ravioli Angelo');
    await expect(page.locator('span:has(input#shipTo)')).toHaveText('Torikatu 38');
});
```

And then I ran it with `npx playwright test --browser firefox --repeat-each 100`

Can you try the same? Does it fail for you? Can it be that there's something else in your env that makes it fail?

**Comment by rolandz83** [comment]
Hey Andrey, thanks for getting back so quick!
I did what you suggested, there is certainly an improvement. I ran it 20 times, it failed only once.

![Screenshot 2022-01-24 172213](https://user-images.githubusercontent.com/78881730/150874953-cbd0f49a-12b0-4846-8f51-d5d46509c0d3.jpg)

same issue element is not stable and then something else intercepts pointer event.

I ran the test again this time 50 times... all pass
I ran the original test 4 fail 1 pass. so yes defiantly something else is effecting...

**Comment by aslushnikov** [comment]
> I ran the original test 4 fail 1 pass. so yes defiantly something else is effecting...

@rolandz83 I'd appreciate if you can dig more and refine the repro so that I can reproduce it locally!

**Comment by rolandz83** [comment]
Hello again, 
I did a lot of clean up today, removed some packages, dependencies from my project, reinstalled pw as if I'm starting a new project. it did improve the results a little. I ran the tests again 100 times and got 86 pass the rest is failing at random, including two navigation failures at page.goto waiting until "load" which If I remember correctly was a previously reported firefox bug ?

Then out of frustration, I deleted node_modules and reinstalled everything again.... and that did the trick... somewhat...
less then 2% of tests are still failing with "element is not stable - waiting..." message

not sure what else I can do, or how I can have you reproduce this... any other tool I can try besides the debugger/inspector  to see what it is doing under the hood ?

**Comment by rolandz83** [comment]
Also I encourage you to run the code once again few hundred times you will eventually get a fail. 
I went back and ran the test again from the new folder per your suggestion yesterday passed 99 times and failed once. 
seems like persistence is the only way to reproduce this :)

Hoping to resolve this as we have long e2e tests currently running with webkit and chromium.... just cant get firefox to work.

**Comment by pavelfeldman** [comment]
@dgozman it reproduces to me, looks like a re-render issue that you planned to fix in 1.20.

**Comment by dgozman** [comment]
Folding into #9855.
